src/server-entry.js

```diff
@@ -182,6 +184,30 @@ export default context => {
 					logFormatter(`data pre-fetch: ${Date.now() - s}ms`);
 					sp = new Date();
 				}
+				const transactionId = router.currentRoute.query?.kiva_transaction_id
+					? numeral(router.currentRoute.query?.kiva_transaction_id).value()
+					: null;
+
+				const data = transactionId ? apolloClient.readQuery({
```

Comment:
Is this new code running on each page load? And where does this query initially get ran? I see it's being pulled from the cache here in the server entry.

Reply:
Yes, this will run on each page load, but it will only try to read data from the cache. If the query has been executed this will return a result, but be undefined otherwise.

Reply:
@christian14b Calls to `readQuery` need to be wrapped in a try/catch because they will throw exceptions if the data isn't present.